OnlineGameStore.BLL/Services/Service.cs

```
@@ -67,48 +72,112 @@ public virtual async Task<PaginatedResponse<TDetailedDto>> GetAsync(
                     .FirstOrDefault(exception => exception is KeyNotFoundException) ?? agg;
 
             if (inner is KeyNotFoundException)
-                return null;
+                throw new NotFoundException("One or more properties in the DTO were not found in the entity.");
 
-            throw;
+            throw new InternalErrorException("An error occurred while mapping the DTO to the entity.");
+        }
+
+        TEntity? addedEntity;
+        try
+        {
+            addedEntity = await _repository.AddAsync(entity);
+        }
+        catch (ArgumentNullException)
+        {
+            throw new ValidationException("Entity cannot be null.");
+        }
+        catch (DbUpdateException)
+        {
+            throw new ConflictException("An error occurred while adding the entity. It may already exist.");
+        }
+        catch (Exception)
+        {
+            throw new InternalErrorException("An unexpected error occurred while adding the entity.");
         }
 
-        var addedEntity = await _repository.AddAsync(entity);
+        if (addedEntity == null)
+            throw new BadRequestException("An unexpected error occurred while adding the entity.");
 
-        return addedEntity == null ? null : _mapper.Map<TReadDto>(addedEntity);
+        return _mapper.Map<TReadDto>(addedEntity);
     }
 
     public virtual async Task<bool> UpdateAsync(Guid id, TCreateDto dto)
```

Why does it return bool? Would not it be better to just throw a corresponding exception if the Update was not successful and propagate the exception up to the user

---

If we examine the tests, we can see that under certain conditions, this method returns `false`, which is handled in the controllers later.

Remaking this method to return void would require changing a lot of logic across all layers and differentiating the conditions for a `false` return differently.
This is better to include in a separate user story about method refactoring

Secondly, this is beyond the scope of user stories, covered by this PR, and it already works great.